[test_rope_1d.py] | (T3K) T3000 e2e tests | t3000-e2e-tests / models_tttv2_rope_modules_full_set_unit_tests [wh_llmbox]

## CI Failure Report

<!-- AUTO-TRIAGE-SUMMARY-START -->
**Summary:** [CI] test_rope_1d_decode_forward_vs_reference fails with ValueError: array dimension mismatch (size X vs 1024) in T3000 RoPE unit tests
<!-- AUTO-TRIAGE-SUMMARY-END -->

**Workflow:** (T3K) T3000 e2e tests
**Job:** t3000-e2e-tests / models_tttv2_rope_modules_full_set_unit_tests [wh_llmbox]

### Regression timeline

- **Last passing commit:** N/A (not provided)
- **First failing commit:** `09a0526ebfe5` — 2026-05-29T07:35:35Z — https://github.com/tenstorrent/tt-metal/actions/runs/26624591673
- **Most recent failing commit:** `039767d00b9a` — 2026-06-12T07:52:20Z — https://github.com/tenstorrent/tt-metal/actions/runs/27402500504

The regression has been present for at least ~2 weeks.

### Failing job URLs (last 2 runs)

- https://github.com/tenstorrent/tt-metal/actions/runs/28154000371/job/83380508587
- https://github.com/tenstorrent/tt-metal/actions/runs/28082348118/job/83144014916

### Failing test path(s)

- `models/common/tests/modules/rope/test_rope_1d.py` — specifically `test_rope_1d_decode_forward_vs_reference[...]` parametrized variants

**Run command:**
```
HF_HUB_OFFLINE=1 HF_HOME=/mnt/MLPerf/huggingface HF_MODEL=meta-llama/Llama-3.1-8B-Instruct pytest --tb=short models/common/tests/modules/rope
```

### Error excerpt (terminal failure from logs)

**Run 1** (commit `039767d00b9a`, 2026-06-12):
```
##[error]test_rope_1d_decode_forward_vs_reference[1x1-b1-hd64-llama3-fused-1x1]

ValueError: all the input array dimensions except for the concatenation axis must match exactly, but along dimension 1, the array at index 0 has size 4096 and the array at index 1 has size 1024
FAILED

##[error]test_rope_1d_decode_forward_vs_reference[1x1-b1-hd128-none-fused-1x1]

ValueError: all the input array dimensions except for the concatenation axis must match exactly, but along dimension 1, the array at index 0 has size 16384 and the array at index 1 has size 1024
FAILED

==== 60 failed, 204 passed, 120 deselected, 1 warning in 103.88s (0:01:43) =====
Status: TESTS FAILED (20 failed)
```

**Run 2** (2026-06-11):
```
##[error]test_rope_1d_decode_forward_vs_reference[1x1-b1-hd64-llama3-fused-1x1]

ValueError: all the input array dimensions except for the concatenation axis must match exactly, but along dimension 1, the array at index 0 has size 4096 and the array at index 1 has size 1024
FAILED

##[error]test_rope_1d_decode_forward_vs_reference[1x1-b1-hd128-none-fused-1x1]

ValueError: all the input array dimensions except for the concatenation axis must match exactly, but along dimension 1, the array at index 0 has size 16384 and the array at index 1 has size 1024
FAILED

==== 60 failed, 204 passed, 120 deselected, 1 warning in 105.14s (0:01:45) =====
Status: TESTS FAILED (20 failed)
```

### Reproduction steps

1. Check out a failing commit (e.g. `039767d00b9a`)
2. Set up the environment with the appropriate TTNN wheel
3. Run:
```bash
HF_HUB_OFFLINE=1 HF_HOME=/mnt/MLPerf/huggingface HF_MODEL=meta-llama/Llama-3.1-8B-Instruct \
  pytest --tb=short models/common/tests/modules/rope
```
4. Observe failures in `test_rope_1d_decode_forward_vs_reference` parametrized variants

### Notes

- The failure is a `ValueError` during numpy array concatenation in the reference comparison path of `test_rope_1d_decode_forward_vs_reference` (line 347 of `test_rope_1d.py`). The RoPE output tensor has a larger sequence dimension (e.g. 4096 or 16384) while the expected reference array has size 1024, indicating a mismatch in how the decode output is being sliced or padded before comparison.
- Both fused and non-fused variants fail; head dimensions hd64 and hd128 both affected.
- 60 total pytest failures across both runs (same count); the CI summary reports 20 distinct failed test names.

**Determinism check:**
- Run 1 terminal error: `ValueError: all the input array dimensions except for the concatenation axis must match exactly, but along dimension 1, the array at index 0 has size 4096 and the array at index 1 has size 1024` — `test_rope_1d_decode_forward_vs_reference[1x1-b1-hd64-llama3-fused-1x1]` FAILED
- Run 2 terminal error: `ValueError: all the input array dimensions except for the concatenation axis must match exactly, but along dimension 1, the array at index 0 has size 4096 and the array at index 1 has size 1024` — `test_rope_1d_decode_forward_vs_reference[1x1-b1-hd64-llama3-fused-1x1]` FAILED
- Both runs: identical test name, identical exception class and message pattern, identical failure count (60 failed, 204 passed).

<!-- AUTO-TRIAGE-METADATA-START -->
`Auto-triage-workflow: (T3K) T3000 e2e tests`
`Auto-triage-job-name: t3000-e2e-tests / models_tttv2_rope_modules_full_set_unit_tests [wh_llmbox]`
<!-- AUTO-TRIAGE-METADATA-END -->

<!-- PIPELINE-REPORT-START -->
## Pipeline report
Analysis report: not found
Slack notification: not found
<!-- PIPELINE-REPORT-END -->

<!-- AUTO-TRIAGE-INFO-START -->
## About this issue (automated CI auto-triage)
This issue was opened and assigned by an automated AI system. It posts through a bot token, so notifications may look like they come from a person, but they are automated.

The owner is selected from recent commit history and job-ownership data. It reflects who is *likely able to fix or route* this, not necessarily who caused it, so it can be imperfect.

**If this was assigned to the wrong person, you can:**
- Comment `REASSIGN OWNER: <why this is outside your area / which team or test-area it belongs to>` and the system will automatically re-route it to a better-suited owner.
- Unassign yourself (this stops notifications to you).
- Close the issue if it is not a real or relevant failure.

The assigned owner is reminded about every 3 days until the issue is resolved. Full guide: https://tenstorrent.atlassian.net/wiki/spaces/MI6/pages/2545221692/Auto-Triage+Info
<!-- AUTO-TRIAGE-INFO-END -->

--- a/models/common/tests/modules/rope/test_rope_1d.py
+++ b/models/common/tests/modules/rope/test_rope_1d.py
@@ -345,8 +345,10 @@ def test_rope_1d_decode_forward_vs_reference(
     assert "decode" in trans_mats
     assert "prefill" in trans_mats
 
-    # Prefill trans mat PCC: TTTv2 uses head_dim x head_dim
-    prefill_ref = get_rot_transformation_mat(dhead=head_dim)  # [1, 1, head_dim, head_dim]
+    # Prefill trans mat PCC: the rotary_embedding_llama op requires a single
+    # TILE_SIZE x TILE_SIZE tile (TT_FATAL on any other shape), so the module
+    # builds the prefill trans-mat at TILE_SIZE, not head_dim. See rope_1d.py.
+    prefill_ref = get_rot_transformation_mat(dhead=ttnn.TILE_SIZE)  # [1, 1, 32, 32]
     prefill_tt = to_torch_auto_compose(trans_mats["prefill"])
     prefill_tt_trimmed = prefill_tt[:1, :1, : prefill_ref.shape[2], : prefill_ref.shape[3]]
     pcc_prefill, msg_prefill = comp_pcc(prefill_ref.to(torch.bfloat16), prefill_tt_trimmed.to(torch.bfloat16), 0.9999)

